File: Algorithm/13Week(2024.04.17 ~/DP/C++/Baekjoon_17291_김동현.cpp
#include <iostream>

using namespace std;
int arr[30];

int main()
{
	int N;
	cin >> N;
	arr[0] = 1; arr[1] = 1;
	for (int i = 1; i < N; i++)
	{
		arr[i + 1] = arr[i] * 2;
		if (i - 3 >= 0 && (i - 2) % 2 == 1) arr[i + 1] -= arr[i - 3];
		if (i - 4 >= 0 && (i - 3) % 2 == 0) arr[i + 1] -= arr[i - 4];
	}
	cout << arr[N];
	return 0;
}

// https://www.acmicpc.net/problem/17291
// 0ms의 위엄

File: Algorithm/13Week(2024.04.17 ~/DP/Python/Baekjoon_17291_김동현.py
N = int(input())
mamoization = [0]*21
mamoization[0] = 1
mamoization[1] = 1
for i in range(2,21):
    if i%2 != 0:
        mamoization[i] = mamoization[i-1]*2
    else:
        mamoization[i] = mamoization[i-1]*2 - (mamoization[i-4] + mamoization[i-5])
print(mamoization[N])

# N = int(input())
# mamoization = {i:0 for i in range(21)}
# death = {i:0 for i in range(25)}
# mamoization[1] = 1
# death[4] = 1
# for i in range(2,N+1):
#     if i%2 == 0:
#         death[i+4] += mamoization[i-1]
#     else:
#         death[i+3] += mamoization[i-1]
#     mamoization[i] = mamoization[i-1]*2 - death[i]
# print(mamoization[N])

# https://www.acmicpc.net/problem/17291
# DP 생각이 안났어